bot/exts/utils/clean.py

```diff
@@ -106,69 +224,84 @@ def predicate_regex(message: Message) -> bool:
+                    await self._delete_messages_individually(messages[current_index:])
+                    if not self.cleaning:
+                        # Means that deletion was canceled while deleting the individual messages
+                        return
+                    break
 
-                    # we could use ID's here however in case if the message we are looking for gets deleted,
-                    # we won't have a way to figure that out thus checking for datetime should be more reliable
-                    if message.created_at < until_message.created_at:
-                        # means we have found the message until which we were supposed to be deleting.
-                        break
+                to_delete.append(message)
 
-                    # Since we will be using `delete_messages` method of a TextChannel and we need message objects to
-                    # use it as well as to send logs we will start appending messages here instead adding them from
-                    # purge.
-                    messages.append(message)
+                if len(to_delete) == 100:
+                    # we can only delete up to 100 messages in a bulk
+                    await channel.delete_messages(to_delete)
+                    to_delete.clear()
 
-                # If the message passes predicate, let's save it.
-                if predicate is None or predicate(message):
-                    message_ids.append(message.id)
+            if len(to_delete) > 0:
+                # deleting any leftover messages if there are any
+                await channel.delete_messages(to_delete)
 
         self.cleaning = False
 
-        # Now let's delete the actual messages with purge.
-        self.mod_log.ignore(Event.message_delete, *message_ids)
-        for channel in channels:
-            if until_message:
-                for i in range(0, len(messages), 100):
-                    # while purge automatically handles the amount of messages
-                    # delete_messages only allows for up to 100 messages at once
-                    # thus we need to paginate the amount to always be <= 100
-                    await channel.delete_messages(messages[i:i + 100])
-            else:
-                messages += await channel.purge(limit=amount, check=predicate)
+        log_messages = []
 
-        # Reverse the list to restore chronological order
-        if messages:
-            messages = reversed(messages)
-            log_url = await self.mod_log.upload_log(messages, ctx.author.id)
+        for messages in message_mappings.values():
+            log_messages.extend(messages)
+
+        if log_messages:
+            # Reverse the list to restore chronological order
+            log_messages = reversed(log_messages)
+            log_url = await self.mod_log.upload_log(log_messages, ctx.author.id)
         else:
             # Can't build an embed, nothing to clean!
             embed = Embed(
```

Comment:
This is a bit out of scope, but I think this section until the end of the function should be exported to a helper function such as `log_message_clean` or whatever name might fit.